airflow/www/views.py

```diff
@@ -2224,20 +2224,27 @@ def trigger(self, dag_id: str, session: Session = NEW_SESSION):
                     "warning",
                 )
 
-        data_interval = dag.timetable.infer_manual_data_interval(run_after=logical_date)
-        if not run_id:
-            run_id = dag.timetable.generate_run_id(
-                logical_date=logical_date,
-                data_interval=data_interval,
-                run_type=DagRunType.MANUAL,
-            )
+        run_after = timezone.utcnow()
+        if logical_date:
+            data_interval = dag.timetable.infer_manual_data_interval(run_after=logical_date)
+            run_after = data_interval.end
+            if not run_id:
+                run_id = dag.timetable.generate_run_id(
+                    logical_date=logical_date,
+                    data_interval=data_interval,
+                    run_type=DagRunType.MANUAL,
+                )
+        else:
+            data_interval = None
+            if not run_id:
+                run_id = f"manual__{timezone.utcnow().isoformat()}"
```

Comment:
```suggestion
        if logical_date:
            data_interval = dag.timetable.infer_manual_data_interval(run_after=run_after)
            run_after = data_interval.end
        else:
            data_interval = None
            run_after = timezone.utcnow()

        if not run_id:
            run_id = dag.timetable.generate_run_id(
                logical_date=logical_date,
                data_interval=data_interval,
                run_type=DagRunType.MANUAL,
            )
```

Let’s not change the run_id part for now; I think @prabhusneha’s PR touches this.

Reply:
@uranusjr Should `run_after` be passed as part of the trigger method, or can it be `timezone.utcnow()`?

Reply:
Sure

Reply:
But this will fail tests. So I will pass logical_date as part of tests then

Reply:
changed it